### PR TITLE
Add 'inplace' ability to rotate.api functions

### PR DESCRIPTION
**Pull request by jmcvey3**
Namely rotate2, set_declination, and set_inst2head_rotmat. I set the default value as True, and added these functions to the velds attribute. Also made sure that particular functions that return a dataset are creating a new dataset to keep the original input intact.

### CODE REVIEW

File: dolfyn/rotate/api.py [hunk 1]
```diff
@@ -265,6 +276,7 @@ def set_inst2head_rotmat(ds, rotmat):
     if not csin == 'beam':  # csin not 'beam', then we're in inst
         ds = r_vec._rotate_inst2head(ds)
     if csin not in ['inst', 'beam']:
-        ds = rotate2(ds, csin)
+        rotate2(ds, csin, inplace=True)
 
-    return ds
```
Comment (lkilcher) [inline dolfyn/rotate/api.py:270]: Same comment as above.

File: dolfyn/rotate/api.py [hunk 2]
```diff
@@ -22,7 +22,7 @@
     'rdi': r_rdi}
 
 
-def rotate2(ds, out_frame='earth'):
+def rotate2(ds, out_frame='earth', inplace=True):
```
Comment (lkilcher) [inline dolfyn/rotate/api.py:25]: I'm torn here... On the one hand, I feel like we want `inplace=False` for these functions. I generally expect functions to return something to me. But we definitely want `inplace=True` for the methods. So maybe this inconsistency between those two places will create confusion?

File: dolfyn/velocity.py [hunk 3]
```diff
@@ -25,7 +25,7 @@ class Velocity():
     ########
     # Major components of the dolfyn-API
 
-    def rotate2(self, out_frame='earth'):
+    def rotate2(self, out_frame='earth', inplace=True):
```
Comment (lkilcher) [inline dolfyn/velocity.py:28]: Whatever we do for the functions in `rotate.api`, we definitely `inplace=True` for all of these methods.

**Review by lkilcher — COMMENTED** [review]
Take a look at the comments, and let me know what you think?

File: dolfyn/rotate/api.py [hunk 4]
```diff
@@ -101,7 +102,8 @@ def rotate2(ds, out_frame='earth'):
             func = getattr(rmod, '_' + rc[inow] + '2' + rc[inow + 1])
         ds = func(ds, reverse=reverse)
 
-    return ds
```
[the remark this answers is not in the document]
Comment (lkilcher) [reply dolfyn/rotate/api.py:104]: I've also come around to the idea that its better to return `None` when `inplace=True`. So disregard this.

File: dolfyn/rotate/api.py (hunk 2 above)
Comment (lkilcher) [reply dolfyn/rotate/api.py:25]: After thinking about this more, I think it's best to leave these consistent with the methods. So `inplace=True` is the right way to go here.

**Comment by lkilcher** [comment]
Do we need a note in the functions that have the `inplace` option along the lines of:

> When making these kinds of inplace changes to the data, this only changes the data in memory not the data on disk. You will need to use a `.save` method in order to write these changes to disk.

Or is this obvious/unnecessary? ... I just don't know how much `xarray` users expect the data in memory to be identifcal to what is on disk.

**Comment by jmcvey3** [comment]
> Do we need a note in the functions that have the `inplace` option along the lines of:
> 
> > When making these kinds of inplace changes to the data, this only changes the data in memory not the data on disk. You will need to use a `.save` method in order to write these changes to disk.
> 
> Or is this obvious/unnecessary? ... I just don't know how much `xarray` users expect the data in memory to be identifcal to what is on disk.

Not entirely following, but I'm also not entirely familiar with this sort of thing. If you open up a netcdf file, anything you do it must be saved by the user; it's no different from a word doc: if you don't save your progress and power goes out you'll lose it.

**Comment by lkilcher** [comment]
Great! In that case I think this PR is almost ready to merge. The only Q I have is whether there is a way to 'run the checks' here before we do so?

**Pull request merged**